### ISSUE TITLE
errno for error codes for key management break on newlib

### ISSUE THREAD

**Issue by galak** [zephyrproject-rtos/zephyr#10878]
The following added errno values aren't defined by newlib and thus cause issues when building with newlib:

```
/* specific encryption errno values */
#define ENOKEY 126       /* Required key not available */
#define EKEYEXPIRED 127  /* Required key not available */
#define EKEYREVOKED 128  /* Key has been revoked */
#define EKEYREJECTED 129 /* Key was rejected by service */
```

**Comment by galak** [comment]
@nilsMontenegro can you look at this.

**Comment by nilsMontenegro** [comment]
Yes. Can we define them in newlib as well? Would that be a pull request to the zephyr-SDK?

**Comment by galak** [comment]
> Yes. Can we define them in newlib as well? Would that be a pull request to the zephyr-SDK?

defining them in newlib isn't an option.  We'd need a zephyr specific header for such extensions if we want to keep these defines.

**Comment by galak** [comment]
This has been partially addressed by PR #10881, and the rest with PR #10879

**Comment by nilsMontenegro** [comment]
@galak Yes, with these PRs in, that will not be a problem any more.
